gateway/src/apicast/policy/routing/apicast-policy.json

```diff
@@ -0,0 +1,154 @@
+      "operation": {
+        "type": "object",
+        "$id": "#/definitions/operation",
+        "properties": {
+          "thing_to_match": {
+            "type": "string",
+            "enum": [
+              "path",
+              "header",
+              "query_arg",
+              "jwt_claim"
+            ]
+          },
+          "op": {
+            "type": "string",
+            "enum": [
+              "==",
+              "!=",
+              "matches"
+            ]
+          },
+          "value": {
+            "type": "string"
+          },
+          "value_type": {
+            "description": "How to evaluate 'type'",
+            "type": "string",
+            "default": "plain",
+            "oneOf": [
+              {
+                "enum": ["plain"],
+                "title": "Evaluate 'value' as plain text."
+              },
+              {
+                "enum": ["liquid"],
+                "title": "Evaluate 'value' as liquid."
+              }
+            ]
+          }
+        },
+        "required": [
+          "thing_to_match",
+          "op",
+          "value"
+        ],
+        "dependencies": {
+          "thing_to_match": {
+            "oneOf": [
+              {
+                "properties": {
+                  "thing_to_match": {
+                    "enum": [
+                      "header"
+                    ]
+                  },
+                  "header_name" : {
+                    "type": "string"
+                  }
+                },
+                "required": [
+                  "header_name"
+                ]
+              },
+              {
+                "properties": {
+                  "thing_to_match": {
+                    "enum": [
+                      "query_arg"
+                    ]
+                  },
+                  "query_arg_name" : {
+                    "type": "string"
+                  }
+                },
+                "required": [
+                  "query_arg_name"
+                ]
+              },
+              {
+                "properties": {
+                  "thing_to_match": {
+                    "enum": [
+                      "jwt_claim"
+                    ]
+                  },
+                  "jwt_claim_name" : {
+                    "type": "string"
+                  }
+                },
+                "required": [
+                  "jwt_claim_name"
+                ]
+              },
+              {
+                "properties": {
+                  "thing_to_match": {
+                    "enum": [
+                      "path"
+                    ]
+                  }
+                }
+              }
+            ]
+          }
+        }
+      }
+    },
+    "properties": {
+      "rules": {
+        "description": "list of rules to be applied",
+        "type": "array",
+        "items": {
+          "type": "object",
+          "properties": {
+            "url": {
```

Comment:
Do we need just this? We don't need to control whether it rewrites Host header?

Reply:
I'm not sure. Do you think we need a way to prevent rewriting the Host header?
Currently, the host is always rewritten calling this method:
https://github.com/3scale/apicast/blob/75eb7e77d466d5ac6ba0aded9446838c87414e95/gateway/src/apicast/upstream.lua#L153

Reply:
I think we should double check with out requirements and with people using this. 
Host header can be different from the hostname where the traffic should go. 
There might be a need to supply own Host header in some cases.

Reply:
I think that we can leave this as it is and add the extra parameter later if needed.
Let's check with @vramosp

Reply:
Is it feasible to offer the following set of options?

-  Host header overwrite by new upstream host (default)
-  Host header unchanged
-  Host header manual overwrite with a new fixed value

Reply:
I created an issue for this #981